models/recovery_code.go

```diff
@@ -0,0 +1,51 @@
+package models
+
+import (
+	"database/sql"
+	"github.com/gofrs/uuid"
+	"github.com/netlify/gotrue/storage"
+	"github.com/pkg/errors"
+	"time"
+)
+
+type RecoveryCode struct {
+	ID           uuid.UUID  `json:"id" db:"id"`
+	UserID       uuid.UUID  `json:"user_id" db:"user_id"`
+	CreatedAt    *time.Time `json:"created_at" db:"created_at"`
+	RecoveryCode string     `json:"recovery_code" db:"recovery_code"`
+	UsedAt   *time.Time `json:"used_at" db:"used_at"`
+}
+
+func (RecoveryCode) TableName() string {
+	tableName := "mfa_recovery_codes"
+	return tableName
+}
+
+// Returns a new recovery code associated with the user
+func NewRecoveryCode(user *User, recoveryCode string, now *time.Time) (*RecoveryCode, error) {
+
+	id, err := uuid.NewV4()
+	if err != nil {
+		return nil, errors.Wrap(err, "Error generating unique id")
+	}
+	code := &RecoveryCode{
+		ID:           id,
+		UserID:       user.ID,
+		RecoveryCode: recoveryCode,
+		CreatedAt:    now,
```

Comment:
gobuffalo will automatically set this for you (https://gobuffalo.io/documentation/database/models/#unix-timestamps)

Reply:
Ah yup this is intentionally done to skirt Buffalo's default behaviour so that each batch of recovery codes has the same timestamp and users can query batches of codes by timestamp. Otherwise there would be slight difference between the timestamp recorded but it's a good point -- I'm not sure if :

1. The time difference is significant since it's probably on the order of ms or lower
2. It might be better to have an explicit timestamp field

Let me check what the timestamps in DB are and adjust accordingly

Reply:
Yeah reverted so we use the default created by Buffalo -- users can check the audit log to tally timestamps if they want to look into when a batch was created